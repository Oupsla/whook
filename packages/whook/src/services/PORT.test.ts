import initPORT from './PORT';

describe('initPORT', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should use the env port first', async () => {
    const port = await initPORT({
      ENV: { PORT: '1337' },
      log,
    });

    expect({
      port,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should find a port by itself if no env port', async () => {
    const port = await initPORT({
      log,
    });

    expect(port);
    expect({
      logCalls: log.mock.calls
        .filter(args => 'stack' !== args[0])
        .map(([arg1, arg2, ...args]) => {
          return [arg1, arg2.replace(/port (\d+)/, 'port ${PORT}'), ...args];
        }),
    }).toMatchSnapshot();
  });
});
