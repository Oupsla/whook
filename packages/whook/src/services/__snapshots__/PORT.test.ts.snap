// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initPORT should find a port by itself if no env port 1`] = `
Object {
  "logCalls": Array [
    Array [
      "debug",
      "üè≠ - Initializing the PORT service.",
    ],
    Array [
      "warning",
      "‚úî - Found a free port \${PORT}",
    ],
  ],
}
`;

exports[`initPORT should use the env port first 1`] = `
Object {
  "logCalls": Array [
    Array [
      "debug",
      "üè≠ - Initializing the PORT service.",
    ],
    Array [
      "warning",
      "‚ôªÔ∏è - Using ENV port 1337",
    ],
  ],
  "port": 1337,
}
`;
